### PR TITLE
NCL-6600 Add API for using latest version

### PR DESCRIPTION
**Pull request by rnc**

### CODE REVIEW

File: core/src/main/java/org/commonjava/maven/ext/core/impl/RESTCollector.java [hunk 1]
```diff
@@ -85,15 +84,15 @@ private void collect( final List<Project> projects )
             return;
         }
 
-        final ArrayList<ProjectVersionRef> newProjectKeys = new ArrayList<>();
+        final ArrayList<ProjectVersionRef> restLookupProjectVersionParamList = new ArrayList<>();
```
Comment (dwalluck) [inline core/src/main/java/org/commonjava/maven/ext/core/impl/RESTCollector.java:87]: Should be `List'.

**Comment by rnc** [comment]
@dwalluck Did you have any last comments on this else I'll merge it tomorrow

File: io/src/main/java/org/commonjava/maven/ext/io/rest/DefaultTranslator.java [hunk 2]
```diff
@@ -485,7 +455,7 @@ else if (originalBody.startsWith( "javax.validation.ValidationException: " )) {
             }
         }
 
-        public List<Task> split()
+        public List<Task> split( ENDPOINT endpointType )
```
[the remark this answers is not in the document]
Comment (rnc) [reply io/src/main/java/org/commonjava/maven/ext/io/rest/DefaultTranslator.java:458]: We don't have commons-collection so I'll resolve this.

File: io/src/main/java/org/commonjava/maven/ext/io/rest/Translator.java [hunk 3]
```diff
@@ -41,12 +40,16 @@
...
-    // lookupLatestVersions
-    List<ProjectVersionRef> findBlacklisted( ProjectRef project ) throws RestException;
+    Map<ProjectVersionRef, String> lookupVersions( List<ProjectVersionRef> projects ) throws RestException;
 
+    /**
+     * Executes HTTP request to a REST service that translates versions. While similar to {@link Translator#lookupVersions(List)}
+     * for this version, the DependencyAnalyser will ignore suffix priority and return the latest version for the
+     * configured suffix mode. This is typically used for project version lookups.
+     *
+     * @param projects - List of projects (GAVs)
```
[the remark this answers is not in the document]
Comment (rnc) [reply io/src/main/java/org/commonjava/maven/ext/io/rest/Translator.java:50]: Thanks

File: io/src/main/java/org/commonjava/maven/ext/io/rest/DefaultTranslator.java [hunk 4]
```diff
@@ -58,13 +57,19 @@
     private static final GenericType<List<MavenLookupResult>> lookupType = new GenericType<List<MavenLookupResult>>()
     {
     };
-    private static final GenericType<List<ProjectVersionRef>> pvrTyoe = new GenericType<List<ProjectVersionRef>>()
-    {
-    };
 
-    private static final String LOOKUP_GAVS = "lookup/maven";
+    public enum ENDPOINT {
```
[the remark this answers is not in the document]
Comment (rnc) [reply io/src/main/java/org/commonjava/maven/ext/io/rest/DefaultTranslator.java:61]: Fair point.

File: io/src/test/java/org/commonjava/maven/ext/io/rest/handler/AddSuffixJettyHandler.java [hunk 5]
```diff
@@ -97,7 +97,7 @@ public void handle( String target, Request baseRequest, HttpServletRequest reque
             }
         }
 
-        if ( target.startsWith( this.endpoint ) )
+        if ( target.contains( this.endpoint ) )
```
[the remark this answers is not in the document]
Comment (rnc) [reply io/src/test/java/org/commonjava/maven/ext/io/rest/handler/AddSuffixJettyHandler.java:101]: I've removed the `/` from the AddSuffixJettyHandler but I'm leery of changing the core code just to make the tests look nicer.

File: io/src/main/java/org/commonjava/maven/ext/io/rest/DefaultTranslator.java [hunk 6]
```diff
@@ -58,13 +57,19 @@
...
+        LOOKUP_GAVS ("lookup/maven"),
+        LOOkUP_LATEST ("lookup/maven/latest");
```
[the remark this answers is not in the document]
Comment (rnc) [reply io/src/main/java/org/commonjava/maven/ext/io/rest/DefaultTranslator.java:64]: Due to the code on line 122 in the ctor to handle whether the `restURL` has a trailing slash or not I think its simpler to keep it as it is.

**Comment by rnc** [comment]
For future reference, I have entered https://github.com/release-engineering/pom-manipulation-ext/pull/846 to resolve comments raised after merge.

**Pull request merged**